File: goimp/components/walls.go
package components

import "fmt"

type Wall interface {
	MapSite
}

type SimpleWall struct{}

func CreateSimpleWall() Wall {
	return new(SimpleWall)
}

func createSimpleWall() *SimpleWall {
	return new(SimpleWall)
}

func (w SimpleWall) Enter() {
	fmt.Println("crossing a simple walls")
}

func (w SimpleWall) String() string {
	return "SimpleWall{}"
}

type BombedWall struct {
}

func CreateBombedWall() Wall {
	return new(BombedWall)
}

func createBombedWall() *BombedWall {
	return new(BombedWall)
}

func (w BombedWall) Enter() {
	fmt.Println("crossing a bombed walls")
}

func (w BombedWall) String() string {
	return "BombedWall{}"
}

File: py/patterns/components/walls.py
from .common import MapSite


class Wall(MapSite):
    def enter(self):
        pass

    def __str__(self) -> str:
        return "simple wall{}"


class BombedWall(Wall):
    def enter(self):
        pass

    def __str__(self):
        return "bombed wall{}"
